test/spec/duplexSpec.js

```diff
@@ -407,7 +413,7 @@ describe("duplex", function () {
           obj.foo = "something";
 
           var patches = jsonpatch.generate(observer);
-          expect(patches).toEqual([{op: 'replace', path: '/foo', value: "something"}]);
+          expect(patches).toEqual([{op: 'add', path: '/foo', value: "something"}]);
```

Comment:
>should generate replace, when `undefined` property is set to something
Yeah but for me it's strange, for example object.property = undefined, but if we doesn't assign to it `undefined` and write into console object.property, we also receive undefined.

So it's definatelly add instead of update